dashboard-ui/src/pages/home.tsx

```diff
@@ -196,6 +196,44 @@ function useLogFileInfo(uids: string[], ownershipMap: Map<string, string[]>) {
   return logFileInfo;
 }
 
+function useFilteredWorkloads() {
+  const { kubeContext, search, namespace } = useContext(Context);
+
+  const cronjobs = useCronJobs(kubeContext);
+  const daemonsets = useDaemonSets(kubeContext);
+  const deployments = useDeployments(kubeContext);
+  const jobs = useJobs(kubeContext);
+  const pods = usePods(kubeContext);
+  const replicasets = useReplicaSets(kubeContext);
+  const statefulsets = useStatefulSets(kubeContext);
+
+  const filterCronJobs = applySearchAndFilter(cronjobs.fetching, cronjobs.data?.batchV1CronJobsList?.items, search, namespace);
+  const filterDaemonsets = applySearchAndFilter(daemonsets.fetching, daemonsets.data?.appsV1DaemonSetsList?.items, search, namespace);
+  const filterPods = applySearchAndFilter(pods.fetching, pods.data?.coreV1PodsList?.items, search, namespace);
+  const filterJobs = applySearchAndFilter(jobs.fetching, jobs.data?.batchV1JobsList?.items, search, namespace);
+  const filterDeployments = applySearchAndFilter(deployments.fetching, deployments.data?.appsV1DeploymentsList?.items, search, namespace);
+  const filterReplicasets = applySearchAndFilter(replicasets.fetching, replicasets.data?.appsV1ReplicaSetsList?.items, search, namespace);
+  const filterStatefulsets = applySearchAndFilter(statefulsets.fetching, statefulsets.data?.appsV1StatefulSetsList?.items, search, namespace);
+
+  return {
+    cronjobs: cronjobs.data?.batchV1CronJobsList?.items,
+    daemonsets: daemonsets.data?.appsV1DaemonSetsList?.items,
+    deployments: deployments.data?.appsV1DeploymentsList?.items,
+    jobs: jobs.data?.batchV1JobsList?.items,
+    pods: pods.data?.coreV1PodsList?.items,
+    replicasets: replicasets.data?.appsV1ReplicaSetsList?.items,
+    statefulsets: statefulsets.data?.appsV1StatefulSetsList?.items,
+
+    filterCronJobs,
+    filterDaemonsets,
+    filterDeployments,
+    filterJobs,
+    filterPods,
+    filterReplicasets,
+    filterStatefulsets,
+  };
+}
+
```

Comment:
the logic works fine. but the issue is we are still doing the filtering, when we don't need it.
you could add some conditionals inside the hook to filter only when required and return the original data, when filtering isn't required. somethink like below might work.

```
  const { kubeContext, search, namespace, workloadFilter } = useContext(Context);

const shouldFilter = (workloadType: Workload) => workloadFilter === undefined || workloadFilter === workloadType;

  const filterCronJobs = shouldFilter(Workload.CRONJOBS)
    ? applySearchAndFilter(cronjobs.fetching, cronjobs.data?.batchV1CronJobsList?.items, search, namespace)
    : cronjobs.data?.batchV1CronJobsList?.items;
```